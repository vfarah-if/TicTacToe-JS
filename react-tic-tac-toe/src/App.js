import React from 'react';
import './App.css';
import Game from './components/Game';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <h1>React tic-tac-toe</h1>        
        <section>
          <Game/>
        </section>
      </header>
    </div>
  );
}

export default App;
