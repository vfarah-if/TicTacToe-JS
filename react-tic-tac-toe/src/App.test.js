import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { shallow } from 'enzyme';
import ShallowRenderer from 'react-test-renderer/shallow';

import App from './App';

function gameElement(wrapper) {
  return wrapper.find('Game');
}

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    render(<App />, div);
    unmountComponentAtNode(div);
  });
  
  it('should render a game using enzyme', () => {
    const wrapper = shallow(<App />);
    const element = gameElement(wrapper);
    expect(element).toBeTruthy();
  });

  it('should render a game using shallow renderer', () => {
    const renderer  = new ShallowRenderer();
    renderer.render(<App />);
    const result = renderer.getRenderOutput();

    expect(result.props.children).toMatchSnapshot();
  });
});

