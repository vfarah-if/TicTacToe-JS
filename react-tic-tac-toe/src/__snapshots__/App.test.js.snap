// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render a game using shallow renderer 1`] = `
<header
  className="App-header"
>
  <h1>
    React tic-tac-toe
  </h1>
  <section>
    <Game />
  </section>
</header>
`;
